### DISCUSSION TITLE
Considering removing game images

### DISCUSSION THREAD

**Discussion by jongrim** [Playabl-io/playabl#222] [Announcements]
Hello all,

I am posting this to collect feedback on game images to help me decide if they should continue to be supported.

_To be clear, this only affects the game image shown when browsing games and on the game homepage. Community images and user avatars are not being considered for removal._

**Why remove images?**
Playabl uses [supabase](https://supabase.com/) for storing data and files such as images. Currently, Playabl runs on the Pro tier which has certain caps for storage and what is called "image transformations". Since adding image transformations, Playabl has been exceeding the allowed number by a large margin, and supabase has notified me that I must either pay for a higher tier plan, or reduce the number of transformations. If I do not, they have the right to stop the project completely, thus making all of the site inaccessible.

<details>
<summary>quickly, what are image transformations?</summary>
Image transformations take the image that was uploaded by a user, and scale it to a more appropriate size for others. This is necessary because often people do not upload a web optimized image, and having large images costs users their own data and time in loading.
</details>

I'm not willing to pay for a higher tier at this point since Playabl is comfortably running in every other cost area (total storage, auth users, egress, etc), so that leaves the option of removing the image transformations or images all together. Simply removing the transformations is tempting, but leaves the site in a poor usability state where images can be a drain on data and cause layout shift. There are techniques to alleviate this, but some of those require different application architectures that I can't currently support. In short, having the game images be handled well is not a small task, so it comes with a fairly high cost.

**What value do game images bring?**
Not much, IMO, but that's where I want to here from others. In my view, they can be cool to look at, and at best give an idea about the "flavor" or tone of the game. They help make the site design less text heavy which is visually pleasing, but outside these the game image does not serve much use. A player still has to read the game description to understand the game and tone, so it's my theory that little would be lost in removing the image altogether.

But what do you think? Do you think having images for the game is important for one reason or another? Let me know!

**Comment by BigEZ** [comment]
Quick disclaimer, I am very new to this site so take my feedback with a pinch of salt.

I browsed the game page on phone and laptop and agree with the advantages you listed in "What value do game images bring?".
It's eye-catching, and can tell you a lot about the vibe of the game. If you're just window-shopping by going down the list, the picture is what will get you, before the description. I also think, at the moment, people did not grab the value of it and most of the image don't really pop-out or make a game stand-out.

I think it boils down to how you envision the site to be used.
If you want it used by established teams, groups, associations, etc., pictures on that page are really not that important.
If you want people to browse through the list to discover new groups and new games, I think the pictures are pretty important.

In any case, not having pictures is better than not having an app at all, of course.

Did you consider having a default picture at the proper default size when people chose a game system? "You play Cyberpunk, you get a cyberpunk logo / picture. You play Cthullu, you get a Cthullu picture. You play an unkown game system, you upload your picture"? That may add a bit of complexity in development, though.

**Reply by jongrim** [reply]
That is a clever, novel idea. A library of stock imagery would mean I could pre-optimize the images, and caching will work much better since there would be only so many different images to serve. The site can retain having some imagery to keep the design engaging, but it strikes a nice balance with maintenance and performance. This even has me thinking that potentially, if there were interest from the broader community, these images could be rotated with new drawings and images contributed by members. Could be a cool way to highlight their work and maybe bring them some attention.

I wouldn't try and match the system to an image, I don't think. Instead I think I'd replace the choose image step to allow picking from the library of images, and each could be tagged with a category to help filtering. So maybe a goal of around 20 - 30 images to start, covering a range of general motifs like:
- sci-fi
- fantasy
- horror
- cyberpunk
- abstract / surreal
- western
- wuxia

I'm sure there are some important categories I'm overlooking but this is just off the top of my head. I really like this idea though and will keep thinking on it. Thank you for the feedback and suggestion!
